src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -999,4 +1023,553 @@ bool ReturnTypeIsValueTypeOrGeneric(
+      break;
+    case ELEMENT_TYPE_STRING:
+      pEmit->DefineTypeRefByName(corLibRef, SystemString.data(), &token);
+      break;
+    case ELEMENT_TYPE_OBJECT:
+      pEmit->DefineTypeRefByName(corLibRef, SystemObject.data(), &token);
+      break;
+    case ELEMENT_TYPE_CLASS:
+      pbCur++;
+      token = CorSigUncompressToken(pbCur);
+      break;
+    case ELEMENT_TYPE_VALUETYPE:
+      pbCur++;
+      token = CorSigUncompressToken(pbCur);
+      break;
+    case ELEMENT_TYPE_GENERICINST:
+    case ELEMENT_TYPE_SZARRAY:
+    case ELEMENT_TYPE_MVAR:
+    case ELEMENT_TYPE_VAR:
+      pEmit->GetTokenFromTypeSpec(
+          pbCur, length - static_cast<ULONG>(pbCur - pTemp), &token);
+      break;
+    default:
+      break;
+  }
+  return token;
+}
+
+WSTRING GetSigTypeTokName(PCCOR_SIGNATURE& pbCur, const ComPtr<IMetaDataImport2>& pImport) {
+  WSTRING tokenName = ""_W;
+  bool ref_flag = false;
+  if (*pbCur == ELEMENT_TYPE_BYREF) {
+    pbCur++;
+    ref_flag = true;
+  }
+
+  switch (*pbCur) {
+    case ELEMENT_TYPE_BOOLEAN:
+      tokenName = SystemBoolean;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_CHAR:
+      tokenName = SystemChar;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_I1:
+      tokenName = SystemByte;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_U1:
+      tokenName = SystemSByte;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_U2:
+      tokenName = SystemUInt16;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_I2:
+      tokenName = SystemInt16;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_I4:
+      tokenName = SystemInt32;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_U4:
+      tokenName = SystemUInt32;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_I8:
+      tokenName = SystemInt64;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_U8:
+      tokenName = SystemUInt64;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_R4:
+      tokenName = SystemSingle;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_R8:
+      tokenName = SystemDouble;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_I:
+      tokenName = SystemIntPtr;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_U:
+      tokenName = SystemUIntPtr;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_STRING:
+      tokenName = SystemString;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_OBJECT:
+      tokenName = SystemObject;
+      pbCur++;
+      break;
+    case ELEMENT_TYPE_CLASS:
+    case ELEMENT_TYPE_VALUETYPE: {
+      pbCur++;
+      mdToken token;
+      pbCur += CorSigUncompressToken(pbCur, &token);
+      tokenName = GetTypeInfo(pImport, token).name;
+      break;
+    }
+    case ELEMENT_TYPE_SZARRAY: {
+      pbCur++;
+      tokenName = GetSigTypeTokName(pbCur, pImport) + "[]"_W;
+      break;
+    }
+    case ELEMENT_TYPE_GENERICINST: {
+      pbCur++;
+      tokenName = GetSigTypeTokName(pbCur, pImport);
+      tokenName += "["_W;
```

Comment:
nit: Is there a reason you didn't use "<" and ">" to represent the generic instantiations?

Reply:
Because normally in a `ToString` of a generic type it uses `[` instead `<` 

```csharp
Console.WriteLine(typeof(List<int>));
```
```
System.Collections.Generic.List`1[System.Int32]
```

Reply:
Ah makes sense, thanks for clarifying that for me 😄